### PR TITLE
Add Spotify client

### PR DESCRIPTION
**Pull request by alexhorner**
As discussed, a spotify client now exists

### CODE REVIEW

File: api/Silk.Api/Services/SpotifyService.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        
+        public SpotifyService(string clientId, string clientSecret)
+        {
+            _clientId = clientId;
+            _clientSecret = clientSecret;
+            
+            CheckAndRefreshTokens().GetAwaiter().GetResult();
+        }
+        
+        public async Task CheckAndRefreshTokens()
```
Comment (VelvetToroyashi) [inline api/Silk.Api/Services/SpotifyService.cs]: Nitpick: Should be postfixed with `Async`

File: api/Silk.Api/Services/SpotifyService.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        {
+            if (_tokenExpiry == null || _tokenExpiry.Value <= (DateTime.Now - TimeSpan.FromMinutes(5)))
```
Comment (VelvetToroyashi) [inline api/Silk.Api/Services/SpotifyService.cs:38]: Nitpick: Pattern matching (Can't think of how to do it off the top of my head)

File: api/Silk.Api/Services/SpotifyService.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                Response = page!["tracks"]
+            };
+        }
+
+        public async Task<ApiResponse<SpotifyArtistModel>> GetArtist(string artistId)
+        {
+            await CheckAndRefreshTokens();
+            
+            _spotifyHttpClient.DefaultRequestHeaders.Clear();
+            _spotifyHttpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {_accessToken}");
```
Comment (VelvetToroyashi) [inline api/Silk.Api/Services/SpotifyService.cs]: `DefaultRequestHeaders.Authorization.Add("Bearer", _accessToken);`

**Review by VelvetToroyashi — COMMENTED** [review]
LGTM for the most part; see review comments

File: api/Silk.Api/Services/SpotifyService.cs (hunk 2 above)
Comment (alexhorner) [reply api/Silk.Api/Services/SpotifyService.cs:38]: Skipped as discussed

File: api/Silk.Api/Services/SpotifyService.cs (hunk 3 above)
Comment (alexhorner) [reply api/Silk.Api/Services/SpotifyService.cs]: Untested, next commit

File: api/Silk.Api/Services/SpotifyService.cs (hunk 1 above)
Comment (alexhorner) [reply api/Silk.Api/Services/SpotifyService.cs]: Next commit

**Review by VelvetToroyashi — APPROVED** [review]
LGTM

**Pull request merged**